weave/trace_server/errors.py

```diff
@@ -39,15 +65,245 @@ def __init__(self, message: str, api_key_name: str):
+        except json.JSONDecodeError:
+            return {"reason": exc_str}
+
+    def _format_error_to_json_with_extra(
+        self, exc: Exception, extra_fields: dict[str, Any]
+    ) -> dict[str, Any]:
+        """Helper to format exception as JSON or fallback to reason, always adding extra fields."""
+        exc_str = str(exc)
+        try:
+            result = json.loads(exc_str)
+        except json.JSONDecodeError:
+            result = {"reason": exc_str}
+
+        result.update(extra_fields)
+        return result
+
+    def _format_insert_too_large(self, exc: Exception) -> dict[str, Any]:
+        """Format InsertTooLarge exception."""
+        return self._format_error_to_json(exc)
+
+    def _format_transport_query_error(self, exc: Exception) -> dict[str, Any]:
+        """Format TransportQueryError with special permission logic."""
+        transport_exc = exc if isinstance(exc, TransportQueryError) else None
+        if transport_exc and transport_exc.errors:
+            for error in transport_exc.errors:
+                if error.get("extensions", {}).get("code") == "PERMISSION_ERROR":
+                    return {"reason": "Forbidden"}
+                if error.get("message") == "project not found" and error.get(
+                    "path"
+                ) == ["upsertBucket"]:
+                    # This seems counter intuitive, but gorilla returns this error when the project exists
+                    # but the user does not have access to it. This seems like a bug on Gorilla's side.
+                    return {"reason": "Forbidden"}
+        return {"reason": "Forbidden"}
+
+    def _format_missing_llm_api_key(self, exc: Exception) -> dict[str, Any]:
+        """Format MissingLLMApiKeyError with api_key field."""
+        if isinstance(exc, MissingLLMApiKeyError):
+            return self._format_error_to_json_with_extra(
+                exc, {"api_key": exc.api_key_name}
+            )
+        return self._format_error_to_json(exc)
+
+    def _format_object_deleted_error(self, exc: Exception) -> dict[str, Any]:
+        """Format ObjectDeletedError with deleted_at timestamp."""
+        if isinstance(exc, ObjectDeletedError):
+            return self._format_error_to_json_with_extra(
+                exc, {"deleted_at": exc.deleted_at.isoformat()}
+            )
+        return self._format_error_to_json(exc)
+
+
+def get_error_registry() -> ErrorRegistry:
+    """Get the global error registry, initializing it if needed."""
+    global _error_registry
+    if _error_registry is None:
+        _error_registry = ErrorRegistry()
+    return _error_registry
+
+
+def register_error(
+    exception_class: type,
+    status_code: int,
+    formatter: Optional[Callable[[Exception], dict[str, Any]]] = None,
+) -> None:
+    """Convenience function to register an error with the global registry."""
+    get_error_registry().register(exception_class, status_code, formatter)
+
+
+def error_handler(
```

Comment:
Where is this used?

Reply:
nowhere, but I thought it might be more convenient if someone had an error in another module they wanted to keep there, but still register. happy to remove if you think it's unnecessary.